### ISSUE TITLE
How to make KantanaAuthentication example work in WebAPI 2.2?

### ISSUE THREAD

**Issue by rosslynquynhnguyen** [IdentityModel/Thinktecture.IdentityModel#102]
Hi,

I am trying to use IdentityModel in WebAPI 2.2. Inside KatanaAuthentication.Startup class, I need to call:
app.UseWebApi(WebApiConfig.Register());

But in WebAPI 2.2, the signature of Register is:
public static void Register(HttpConfiguration config)

How can I create/get the config obj?

Thank you,
Quynh.

**Comment by leastprivilege** [comment]
well - you need the right authentication middleware - depending how you want to do authentication.

**Comment by rosslynquynhnguyen** [comment]
Hi,

Here is the code sample from the KatanaAuthenticationDemo solution for IdentityModel:
![image](https://cloud.githubusercontent.com/assets/6939516/6097308/2ec589f4-afb1-11e4-9cf1-76052180c0ee.png)

I am trying to follow the example but the signature of WebApiConfig.Register in Web API 2.2 is now:
![image](https://cloud.githubusercontent.com/assets/6939516/6097317/76d9a086-afb1-11e4-848f-d17038ec6e5b.png)

Please note that Register now requires HttpConfiguration config as a parameter, and returns void. Instead of returning an HttpConfiguration object as in Web API 2.1.

Please advise how I can create a HttpConfiguration object to fix this call from OwinStart:
app.UseWebApi(WebApiConfig.Register());

Thank you,
Quynh.

**Comment by rosslynquynhnguyen** [comment]
Hi Dominic,

I have managed to figure it out. A HttpConfiguration object can be easily created. I found the answer in the code example for IdentityServer V3:
https://github.com/IdentityServer/Thinktecture.IdentityServer3.Samples/blob/master/source/MVC%20Authentication/Api/Startup.cs

There is still one issue here, which is the example for IdentityModel does not do your hardwork justice. I intend to write a detailed blog post explaining how to setup/user IdentityModel in common scenarios in the next few weeks.

Thank you,
Quynh.
